lib/view_model/send/send_view_model.dart

```diff
@@ -126,9 +126,14 @@ abstract class SendViewModelBase extends WalletChangeListenerViewModel with Stor
       return '0.00';
     }
 
+    final pricesKeys = fiatConversionStore.prices.keys.toList();
+
+    final selectedCurr = pricesKeys.firstWhereOrNull((element) =>
+    element.title == selectedCryptoCurrency.title && element.tag == selectedCryptoCurrency.tag);
+
     try {
       final fiat = calculateFiatAmount(
-          price: _fiatConversationStore.prices[selectedCryptoCurrency]!,
+          price: fiatConversionStore.prices[selectedCurr]!,
```

Comment:
what's the difference?!
why not directly use `selectedCryptoCurrency`

Reply:
This issue needs further investigation because the condition fiatConversionStore.prices.containsKey(selectedCryptoCurrency) is true when sending ERC20 tokens but false during the exchange flow for ERC20 tokens only. In both cases, an instance of Currency is in the prices map.